### PR TITLE
feat(Typescript): Upgrade to 3.6.3

### PR DESCRIPTION
**Pull request by gcpantazis**
- Upgrades to TS 3.6.3, which necessitated an upgrade to our TS+eslint integration.
- Fixes mostly for `prefer-const`.
- Fix a few typing issues that 3.6.3 was able to flag that 3.4.x missed.

### CODE REVIEW

File: packages/eslint-config-spruce/index.js [hunk 1]
```diff
@@ -52,12 +57,6 @@ module.exports = {
 						}
 					}
 				],
-				'@typescript-eslint/explicit-function-return-type': [
```
Comment (gcpantazis) [inline packages/eslint-config-spruce/index.js]: @kengoldfarb It was easier to just disable this... I figure individual projects can opt into `strict` TS, which would catch all of these issues.

File: packages/react-heartwood-components/src/components/Card/components/CardBuilder.tsx [hunk 2]
```diff
@@ -1,34 +1,28 @@
...
+import Toast, { IToastProps } from '../../Toast/Toast'
+import Card from '../Card'
+import CardBody from './CardBody'
+import CardFooter from './CardFooter'
+import CardHeader, { ICardHeaderProps } from './CardHeader'
+import OnboardingCard, { IOnboardingCardProps } from './OnboardingCard'
+import Scores from './Scores'
 
-export interface ICardBuilderFooter extends Omit<IHWCardFooter, 'buttonGroup'> {
+export interface ICardBuilderFooter extends IHWCardBuilderFooter {
```
Comment (gcpantazis) [inline packages/react-heartwood-components/src/components/Card/components/CardBuilder.tsx]: @liquidg3 This is the type change I mentioned in Slack

File: packages/spruce-node/index.ts [hunk 3]
```diff
@@ -652,7 +652,7 @@ export default class Sprucebot {
 
 		// not found, create it
 		if (!meta) {
-			meta = await this.createMeta(key, value, Array.from(arguments)[2])
```
Comment (gcpantazis) [inline packages/spruce-node/index.ts]: @liquidg3 @kengoldfarb can you sanity check this? At a blush `Array.from(arguments)[2]` would always get options...  but maybe I'm missing something.

File: packages/spruce-skill-server/lib/GraphQLSubscriptionServer.ts [hunk 4]
```diff
@@ -102,6 +102,9 @@ export default class GQLSubscriptionServer {
 		const token = authorizationHeader.replace('JWT ', '')
 		const decoded = jwt.verify(
 			token,
+			// TODO: config's type doesn't describe this API correctly...
```
Comment (gcpantazis) [inline packages/spruce-skill-server/lib/GraphQLSubscriptionServer.ts]: @kengoldfarb Not sure what the deal is here, but noticed that the types file hasn't been updated in npm in a few years. Possibly just totally wrong. We forked this right?

File: packages/react-heartwood-components/src/components/Card/components/CardBuilder.tsx (hunk 2 above)
Comment (liquidg3) [reply packages/react-heartwood-components/src/components/Card/components/CardBuilder.tsx]: fixed.

File: packages/spruce-node/index.ts (hunk 3 above)
Comment (liquidg3) [reply packages/spruce-node/index.ts]: yes, this was legacy when there was magic on the signature of that method. your fix is 100%.

File: packages/spruce-skill-server/lib/GraphQLSubscriptionServer.ts (hunk 4 above)
Comment (kengoldfarb) [reply packages/spruce-skill-server/lib/GraphQLSubscriptionServer.ts]: I think we should now be able to just reference it directly: `config.API_KEY` and have it strongly typed

**Pull request merged**